### DISCUSSION TITLE
Output files

### DISCUSSION THREAD

**Discussion by philippemiron** [Parcels-code/Parcels#982] [Q&A] [answered]
Hi everyone,

Is there a way to stop writing output for some particles in function of a custom variable or if the particles exit the domain for example? This could decrease the size of the output and the writing time at the end of each run considerably in some cases.

e.g. On the 10th day of a 100 days simulation, `particle.alive` is now `False`, we could stop saving/output its future positions.

Thanks!

**Comment by reint-fischer** [comment, marked as answer]
Hi Philippe,

you can simply use `particle.delete()`. This will remove the particle from the active particleset, which means that the output will contain only the observations of that particle up until that moment. An example of this is an 'ageing' kernel:
```
def Ageing(particle, fieldset, time):
    particle.age += particle.dt
    if particle.age >= fieldset.maxage:
        particle.delete()
```
In this example, a constant has been added to the fieldset: `fieldset.add_constant('maxage', <time-in-seconds>)`

If your particles exit the domain and throw an out-of-bounds error, you can also implement the following:
```
def delete_particle(particle, fieldset, time):
    particle.delete()

pset.execute(AdvectionRK4,
             runtime=timedelta(days=30),
             dt=3600,
             output_file=output_file,
             recovery={ErrorCode.ErrorOutOfBounds:delete_particle})
```

**Reply by philippemiron** [reply]
Thank you, I was aware of the ErrorOutOfBounds exception but didn't know we could also call the particle.delete() method in a kernel.
